climada/hazard/trop_cyclone/trop_cyclone.py

```diff
@@ -366,46 +366,99 @@ def from_tracks(
+        interpolation.
+
+        Note: Only frequency changes are applied as suggested by Jewson 2022
+        (https://doi.org/10.1007/s00477-021-02142-6). Applying only frequency anyway
+        changes mean intensities and most importantly avoids possible inconsistencies
+        (including possible counting) that may arise from the application of both
+        frequency and intensity changes, as the relatioship between these two is non
+        trivial to resolve.
 
         Parameters
         ----------
-        ref_year : int
-            year between 2000 ad 2100. Default: 2050
-        rcp_scenario : int
-            26 for RCP 2.6, 45 for RCP 4.5, 60 for RCP 6.0 and 85 for RCP 8.5.
-            The default is 45.
-
+        percentile: str
+            percentiles of Knutson et al. 2020 estimates, representing the mode
+            uncertainty in future changes in TC activity. These estimates come from
+            a review of state-of-the-art literature and models. For the 'cat05' variable
+            (i.e. frequency of all tropical cyclones) the 5th, 25th, 50th, 75th and 95th
+            percentiles are provided. For 'cat45' and 'intensity', the provided percentiles
+            are the 10th, 25th, 50th, 75th and 90th. Please refer to the mentioned publications
+            for more details.
+            possible percentiles:
+                '5/10' either the 5th or 10th percentile depending on variable (see text above)
+                '25' for the 25th percentile
+                '50' for the 50th percentile
+                '75' for the 75th percentile
+                '90/95' either the 90th or 95th percentile depending on variable  (see text above)
+            Default: '50'
+        scenario : str
+            possible scenarios:
+                '2.6' for RCP 2.6
+                '4.5' for RCP 4.5
+                '6.0' for RCP 6.0
+                '8.5' for RCP 8.5
+        target_year : int
+            future year to be simulated, between 2000 and 2100. Default: 2050.
         Returns
         -------
         haz_cc : climada.hazard.TropCyclone
             Tropical cyclone with frequencies and intensity scaled according
-            to the Knutson criterion for the given year and RCP. Returns
-            a new instance of climada.hazard.TropCyclone, self is not
+            to the Knutson criterion for the given year, RCP and percentile.
+            Returns a new instance of climada.hazard.TropCyclone, self is not
             modified.
         """
-        chg_int_freq = get_knutson_criterion()
-        scale_rcp_year  = calc_scale_knutson(ref_year, rcp_scenario)
-        haz_cc = self._apply_knutson_criterion(chg_int_freq, scale_rcp_year)
-        return haz_cc
+
+        tc_cc = copy.deepcopy(self)
+
+        sel_cat05 = np.isin(tc_cc.category, [0, 1, 2, 3, 4, 5])
+        sel_cat03 = np.isin(tc_cc.category, [0, 1, 2, 3])
+        sel_cat45 = np.isin(tc_cc.category, [4, 5])
+
```

Comment:
Hello, I started using this improvement/fix for the generation of future TCs and discovered that if `TropCyclone.category == []`, this method does not complain at all and just return the exact same object. 

This happened to me when I wrote/read hdf5 files using a mix of `Hazard.write|from_hdf5` and `TropCyclone.write|from_hdf5` instead of only `TropCyclone.write|from_hdf5`, but this could probably happen in other case and is easily addressed by raising an Error if categories are not set, or at the very least a warning, to inform the user.

Here a suggestion (which can probably be improved):

```suggestion
        sel_cat05 = np.isin(tc_cc.category, [0, 1, 2, 3, 4, 5])
        sel_cat03 = np.isin(tc_cc.category, [0, 1, 2, 3])
        sel_cat45 = np.isin(tc_cc.category, [4, 5])

        if sel_cat05.size + sel_cat03.size + sel_cat45.size == 0:
            raise ValueError("Cyclone categories are missing, cannot apply climate change scenario")
```

Reply:
Thanks for spotting and also for having started using it. I decided to go for the warning.